cocotb/types/range.py

```diff
@@ -0,0 +1,175 @@
+from typing import Any, Iterator, overload
+from collections.abc import Sequence
+from functools import cached_property, lru_cache
+
+
+class Range(Sequence):
+    r"""
+    Variant of :class:`range` with inclusive right bound.
+
+    In Python, :class:`range` and :class:`slice` have a non-inclusive right bound.
+    In both Verilog and VHDL, ranges and arrays have an inclusive right bound.
+    This type mimic's Python's :class:`range` type, but implements HDL-like inclusive right bounds.
+    Also supports :attr:`left`, :attr:`right`, and :attr:`length` attributes as seen in VHDL.
```

Comment:
```suggestion
    This type mimic's Python's :class:`range` type, but implements HDL-like inclusive right bounds,
    and also supports :attr:`left`, :attr:`right`, and :attr:`length` attributes as seen in VHDL.
```

Reply:
```suggestion
    This type mimics Python's :class:`range` type, but implements HDL-like inclusive right bounds,
    and also supports :attr:`left`, :attr:`right`, and :attr:`length` attributes as seen in VHDL.
```

Reply:
It wasn't happy about me trying to apply multiple fixups to the same line.